pkg/operator/ceph/cluster/controller.go

```diff
@@ -780,6 +782,53 @@ func (c *ClusterController) onDeviceCMUpdate(oldObj, newObj interface{}) {
 // Delete event functions
 // ************************************************************************************************
 
+func (c *ClusterController) dataDirHostPathCleanUp(clust *cephv1.CephCluster) {
+	logger.Infof("starting data dir host path clean up for cluster - %+v", clust)
+	command := []string{"rm", "rf", clust.Spec.DataDirHostPath}
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	validNodes := k8sutil.GetValidNodes(clust.Spec.Storage, c.context.Clientset, cephv1.GetOSDPlacement(clust.Spec.Placement))
+	logger.Infof("valid nodes - %+v", validNodes)
+	for _, node := range validNodes {
+		logger.Infof("cleaning up hostpath for node  %q", node.Name)
+		myVolume := v1.Volume{Name: "test", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: clust.Spec.DataDirHostPath}}}
+		volumes = append(volumes, myVolume)
+		myVolumeMount := v1.VolumeMount{Name: "test", MountPath: clust.Spec.DataDirHostPath}
+		volumeMounts = append(volumeMounts, myVolumeMount)
+		podSpec := v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: k8sutil.TruncateNodeName("dataDirHostPathCleanUp", node.Name),
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{Name: "test", Image: clust.Spec.CephVersion.Image, VolumeMounts: volumeMounts, Command: command}},
+				Volumes:    volumes,
+			},
+		}
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: node.Name}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      k8sutil.TruncateNodeName("dataDirHostPathCleanUp", node.Name),
+				Namespace: clust.Namespace,
+				Labels: map[string]string{
+					k8sutil.AppAttr:     "dataDirHostPathCleanUp",
+					k8sutil.ClusterAttr: clust.Namespace,
+				},
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		logger.Infof("data dir host path clean up Job - %+v", job)
+
+		_, err := c.context.Clientset.BatchV1().Jobs(clust.Namespace).Create(job)
```

Comment:
Here jobs are created, but there is no waiting for them to finish. Is this intentional?